main.cpp

```diff
@@ -164,114 +175,530 @@ good to go!
 
 #include <iostream>
 
+struct IntType;
+struct DoubleType;
+struct FloatType;
+
 struct FloatType
 {
-    float add(float lhs, float rhs);
-    float subtract(float lhs, float rhs);
-    float multiply(float lhs, float rhs);
-    float divide(float lhs, float rhs);
+    float* value;
+    float value_;
+    FloatType(float value_) : value(new float (value_)){}
+    ~FloatType()
+    {
+        delete value;
+        value = nullptr;
+    }
+    FloatType& add(float rhs);
+    FloatType& subtract(float rhs);
+    FloatType& multiply(float rhs);
+    FloatType& divide(float rhs);
+
+    FloatType& add(const FloatType& rhs);
+    FloatType& subtract(const FloatType& rhs);
+    FloatType& muliply(const FloatType& rhs);
+    FloatType& divide(const FloatType& rhs);
+
+    FloatType& add(const DoubleType& rhs);
+    FloatType& subtract(const DoubleType& rhs);
+    FloatType& multiply(const DoubleType& rhs);
+    FloatType& divide(const DoubleType& rhs);
+
+    FloatType& add(const IntType& rhs);
+    FloatType& subtract(const IntType& rhs);
+    FloatType& multiply(const IntType& rhs);
+    FloatType& divide(const IntType& rhs);
+
 };
 struct DoubleType
 {
-    double add(double lhs, double rhs);
-    double subtract(double lhs, double rhs);
-    double multiply(double lhs, double rhs);
-    double divide(double lhs, double rhs);
+    double* value;
+    double value_;
+    DoubleType(double value_) : value(new double(value_)){}
+    ~DoubleType()
+    {
+        delete value;
+        value = nullptr;
+    }
+    DoubleType& add(double rhs);
+    DoubleType& subtract(double rhs);
+    DoubleType& multiply(double rhs);
+    DoubleType& divide(double rhs);
+
+    DoubleType& add(const DoubleType& rhs);
+    DoubleType& subtract(const DoubleType& rhs);
+    DoubleType& multiply(const DoubleType& rhs);
+    DoubleType& divide(const DoubleType& rhs);
+
+    DoubleType& add(const FloatType& rhs);
+    DoubleType& subtract(const FloatType& rhs);
+    DoubleType& multiply(const FloatType& rhs);
+    DoubleType& divide(const FloatType& rhs);
+
+    DoubleType& add(const IntType& rhs);
+    DoubleType& subtract(const IntType& rhs);
+    DoubleType& multiply(const IntType& rhs);
+    DoubleType& divide(const IntType& rhs);
 };
 struct IntType
 {
-    int add(int lhs, int rhs);
-    int subtract(int lhs, int rhs);
-    int multiply(int lhs, int rhs);
-    int divide(int lhs, int rhs);
+    int* value;
+    int value_;
+    IntType(double value_) : value(new int(value_)){}
+    ~IntType()
```

Comment:
Re-read the instructions about what your class is supposed to own.  You have more member variables than you're supposed to.